Rust/Rust.sublime-syntax

```diff
@@ -803,6 +795,40 @@ contexts:
     - match: '[-+%/*]'
       scope: keyword.operator.arithmetic.rust
 
+  impl-trait:
+    - match: '\bimpl\b'
+      scope: storage.type.impl.rust
+      push: impl-trait-types
+
+  impl-trait-types:
+    - match: '\buse\b'
+      scope: keyword.other.rust
+      push: impl-trait-use
+    - include: types
+    - match: '(?=\S)'
+      pop: true
+
+  impl-trait-use:
+    - match: '<'
+      scope: punctuation.definition.generic.begin.rust
+      set: impl-trait-use-generic-body
+    - match: '(?=\S)'
+      pop: true
+
+  impl-trait-use-generic-body:
+    - meta_scope: meta.generic.rust
+    - match: '>'
+      scope: punctuation.definition.generic.end.rust
+      pop: true
+    - include: lifetime
+    - match: ','
+      scope: punctuation.separator.rust
+    - match: '\bSelf\b'
```

Comment:
I don't know Rust, so apologies if this is stupid: Is `Self` a keyword or is it like `self` or `cls` variables in Python?

Reply:
lowercase `self` is scoped `variable.language` and is the equivalent of Python.

`Self` is the type of current object. It is scoped `keyword.other` in various other contexts as well.

It should probably receive a unique `storage.type.self` scope, but that's probably another PR.

Reply:
Oh, neat. Thanks for explaining.